MediaBrowser.Model/Drawing/DrawingUtils.cs

```diff
@@ -57,6 +57,52 @@ public static class DrawingUtils
             return new ImageDimensions(newWidth, newHeight);
         }
 
+        /// <summary>
+        /// Scale down to fill box.
+        /// Returns original size if both width and height are null or zero.
+        /// </summary>
+        /// <param name="size">The original size object.</param>
+        /// <param name="fillWidth">A new fixed width, if desired.</param>
+        /// <param name="fillHeight">A new fixed height, if desired.</param>
+        /// <returns>A new size object or size.</returns>
+        public static ImageDimensions ResizeFill(
+            ImageDimensions size,
+            int? fillWidth,
+            int? fillHeight)
+        {
+            // Return original size if input is invalid.
+            if ((fillWidth == null || fillWidth == 0)
+                && (fillHeight == null || fillHeight == 0))
+            {
+                return size;
+            }
+
+            if (fillWidth == null || fillWidth == 0)
+            {
+                fillWidth = 1;
+            }
+
+            if (fillHeight == null || fillHeight == 0)
+            {
+                fillHeight = 1;
+            }
+
+            double widthRatio = (double)size.Width / (double)fillWidth;
+            double heightRatio = (double)size.Height / (double)fillHeight;
+            double scaleRatio = Math.Min(widthRatio, heightRatio);
+
+            // Clamp to current size.
+            if (scaleRatio < 1)
+            {
+                return size;
+            }
+
+            int newWidth = Convert.ToInt32(Math.Ceiling((double)size.Width / scaleRatio));
+            int newHeight = Convert.ToInt32(Math.Ceiling((double)size.Height / scaleRatio));
```

Comment:
```suggestion
            int newWidth = Convert.ToInt32(Math.Ceiling(size.Width / scaleRatio));
            int newHeight = Convert.ToInt32(Math.Ceiling(size.Height / scaleRatio));
```

Reply:
So, division with int is implicitly done using doubles? What would `(int)123 / (int)100` return?